app/src/main/java/com/github/se/cyrcle/ui/review/AllReviewsScreen.kt

```diff
@@ -330,8 +332,16 @@ fun ReviewCard(
     userViewModel: UserViewModel,
     reviewViewModel: ReviewViewModel
 ) {
-  val currentUser by userViewModel.currentUser.collectAsState()
-  val userSignedIn by userViewModel.isSignedIn.collectAsState(false)
+    val currentUser by userViewModel.currentUser.collectAsState()
+    val userSignedIn by userViewModel.isSignedIn.collectAsState(false)
+
+    var ownerReputationScore = 0.0
+    userViewModel.getUserById(
```

Comment:
This goes with the other comment I left you. Notice that in the review card you get the user's username as an argument in "title". This means that the user was already fetched. You can do both extract the username AND the title in the first fetch (line 292).

The implementation I suggest is adding as parameter to the ReviewCard a level of type Double? instead of Double. This will allow you to easily differentiate between calls for the AllReviewsScreen (will pass a nonNull level) rather than a call from the ViewProfileScreen (will pass a null level)

Reply:
Really good idea ! And it made me notice that the ReviewCard of the current user was not showing the correct tag either, maybe that will fix one of the bugs that louis mentioned : 

> The levels of people for the AllReviewsScreen aren't updating for some reason. I'm on lv.6 on the ProfileScreen and lv.0 on AllReviews.